src/utils/proxySet.ts

```diff
@@ -0,0 +1,96 @@
+import { proxy } from 'valtio'
+
+// properies that we don't want to expose to the end-user
+interface InternalProxySet<T> extends Set<T> {
+  data: T[]
+  toJSON: object
+  hasProxy(value: T): boolean
+}
+
+export function proxySet<T>(
+  initialValues: Iterable<T> | null | undefined = []
+): Set<T> {
+  const set: InternalProxySet<T> = proxy({
+    data: Array.from(new Set(initialValues)),
+    has(value) {
+      return this.data.indexOf(value) !== -1
+    },
+    hasProxy(value) {
+      let hasProxy = false
+      if (typeof value === 'object') {
```

Comment:
```suggestion
      if (typeof value === 'object' && value !== null) {
```
Unfortunate JavaScript.

Reply:
nice catch